src/main/java/org/terasology/alterationEffects/speed/GlueAlterationEffect.java

```diff
@@ -0,0 +1,40 @@
+package org.terasology.alterationEffects.speed;
+
+import org.terasology.alterationEffects.AlterationEffect;
+import org.terasology.alterationEffects.AlterationEffects;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.delay.DelayManager;
+
+public class WalkSpeedAlterationEffect implements AlterationEffect {
+
+    private DelayManager delayManager;
+
+    public WalkSpeedAlterationEffect(Context context) {
+        this.delayManager = context.get(DelayManager.class);
+    }
+
+    @Override
+    public void applyEffect(EntityRef instigator, EntityRef entity, float magnitude, long duration) {
+        boolean add = false;
+        GlueComponent glue = entity.getComponent(GlueComponent.class);
+        if (glue == null) {
+            add = true;
+            glue = new WalkSpeedComponent();
```

Comment:
This doesn't work since `glue` is a `GlueComponent` and you're trying to assign a `WalkSpeedComponent` to it. Just need to update this to also use a `GlueComponent`